### PR TITLE
Added querying for blocks from time frame

### PR DESCRIPTION
**Pull request by RogueElement**
for issue #106, added a way to get blocks from a timeframe and added an example endpoint `/api/block/time/day/{MM-DD-YYY}` to get blocks on an specific day,  `/api/block/time/day/` returns blocks from the current day. Wanted a review before proceeding

### CODE REVIEW

File: apiroutes.go [hunk 1]
```diff
@@ -242,6 +245,21 @@ func getAddressCtx(r *http.Request) string {
...
+		today := time.Now()
+		return int(math.Floor(float64(today.Unix() / 86400)))
+	}
+	dateTime, err := time.Parse("01-02-2006", dateString)
+	if err != nil {
+		apiLog.Trace("cannot parse date %s: %s", dateString, err.Error())
+		today := time.Now()
+		return int(math.Floor(float64(today.Unix() / 86400)))
+	}
+	return int(math.Floor(float64(dateTime.Unix() / 86400)))
```
Comment (chappjc) [inline apiroutes.go]: I think the math.Floor is unnecessary since the division is two integers.

File: apiroutes.go [hunk 2]
```diff
@@ -803,6 +821,36 @@ func (c *appContext) getBlockRangeSteppedSummary(w http.ResponseWriter, r *http.
 	fmt.Fprintf(w, "]")
 }
 
+func (c *appContext) getBlockSummaryByDate(w http.ResponseWriter, r *http.Request) {
+	date := getDateCtx(r)
+	blocks, er := c.BlockData.GetBlockSummaryByTime(date*86400, (date+1)*86400)
```
Comment (chappjc) [inline apiroutes.go]: Might as well keep the int64 all over and keep the units in seconds.

Also, and alt solution to day rounding:  https://play.golang.org/p/9jVLG6vQci

File: dcrsqlite/sqlite.go [hunk 3]
```diff
@@ -596,6 +597,45 @@ func (db *DB) RetrieveBlockSizeRange(ind0, ind1 int64) ([]int32, error) {
 	return blockSizes, nil
 }
 
+// RetrieveBlockSummaryRangeTime returns an array of apitypes.BlockDataBasic for
+// blocks in given time frame
+func (db *DB) RetrieveBlockSummaryRangeTime(start, end int64) ([]*apitypes.BlockDataBasic, error) {
+	N := start - end
+	if N == 0 {
```
Comment (chappjc) [inline dcrsqlite/sqlite.go]: Also check for end<start

File: dcrsqlite/sqlite.go [hunk 4]
```diff
@@ -596,6 +597,45 @@ func (db *DB) RetrieveBlockSizeRange(ind0, ind1 int64) ([]int32, error) {
...
+		return []*apitypes.BlockDataBasic{}, nil
+	}
+
+	blocks := []*apitypes.BlockDataBasic{}
```
Comment (chappjc) [inline dcrsqlite/sqlite.go]: Will also need a limit on max blocks

Comment (chappjc) [reply dcrsqlite/sqlite.go]: So we should estimate as days*288

File: dcrsqlite/sqlite.go [hunk 5]
```diff
@@ -596,6 +597,45 @@ func (db *DB) RetrieveBlockSizeRange(ind0, ind1 int64) ([]int32, error) {
...
+		return nil, err
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		bd := new(apitypes.BlockDataBasic)
+		if err = rows.Scan(&bd.Height, &bd.Size, &bd.Hash, &bd.Difficulty,
+			&bd.StakeDiff, &bd.Time, &bd.PoolInfo.Size, &bd.PoolInfo.Value,
+			&bd.PoolInfo.ValAvg); err != nil {
+			log.Errorf("Unable to scan for sdiff fields: %v", err)
```
Comment (chappjc) [inline dcrsqlite/sqlite.go]: if scan errors, I think we should `continue` instead of appending an empty bd

**Comment by raedah** [comment]
@RogueElement Closing this since what we will summing the totals of all blocks in the range, rather then displaying a list of blocks for the range.

**Pull request closed without merging**